### PR TITLE
lots of matching involving misc racers

### PR DESCRIPTION
**Pull request by FazanaJ**

### CODE REVIEW

File: src/vehicle_misc.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    u8 pad10[0x108];
+    s32 unk118;
+} unk8005C2F0;
+
+void update_tricky(s32 updateRate, f32 updateRateF, Object *obj, Object_Racer *racer, u32 *input, u32 *buttonsPressed, s32 *startTimer);
+void update_bluey(s32 updateRate, f32 updateRateF, Object *obj, Object_Racer *racer, u32 *input, u32 *buttonsPressed, s32 *startTimer);
+void update_bubbler(s32 updateRate, f32 updateRateF, Object *obj, Object_Racer *racer, u32 *input, u32 *buttonsPressed, s32 *startTimer);
+void update_smokey(s32 updateRate, f32 updateRateF, Object *obj, Object_Racer *racer, u32 *input, s32 *stickMag, s32 *startTimer);
+void update_wizpig(s32 updateRate, f32 updateRateF, Object* obj, Object_Racer* racer, u32* input, u32* buttonsPressed, s32* startTimer);
+void update_rocket(s32 updateRate, f32 updateRateF, Object* obj, Object_Racer* racer, u32* input, u32* buttonsPressed, s32* startTimer);
```
Comment (DavidSM64) [inline src/vehicle_misc.h:25]:
```suggestion
void update_wizpig(s32 updateRate, f32 updateRateF, Object *obj, Object_Racer *racer, u32 *input, u32 *buttonsPressed, s32 *startTimer);
void update_rocket(s32 updateRate, f32 updateRateF, Object *obj, Object_Racer *racer, u32 *input, u32 *buttonsPressed, s32 *startTimer);
```

File: src/particles.h [hunk 2]
```diff
@@ -114,31 +325,34 @@ void func_800AEF88(unk800AF024 *arg0, Vertex **arg1, Triangle **arg2);
...
+void func_800AF52C(Object *obj, s32 arg1);
+void func_800AF134(Particle *arg0, s32 arg1, s32 arg2, s16 arg3, s16 arg4, s16 arg5);
 
-void func_800AF134(Object *arg0, s32 arg1, s32 arg2, s16 arg3, s16 arg4, s16 arg5); // Non Matching
 void func_800AF404(s32 arg0); // Non Matching
 void func_800AFC3C(Object *, s32); // Non Matching
 void func_800AE728(s32 arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, s32 arg5); // Non Matching
 void func_800AF714(Object*, s32); // Non matching
-Object* func_800B1130(s32, void*);
 Object* func_800B0BAC();
```
Comment (DavidSM64) [inline src/particles.h:350]:
```suggestion
Object *func_800B0BAC();
```

File: src/vehicle_bluey.c [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    s16 sp5A;
+    f32 xDiff;
+    f32 zDiff;
+    s32 var_v0;
+    s32 sp48;
+    ObjectModel *objModel;
+    Object_68* obj68;
+    s32 sp3C;
+    UNUSED s32 pad;
+    Object* firstRacerObj;
```
Comment (DavidSM64) [inline src/vehicle_bluey.c:79]:
```suggestion
    ObjectModel *objModel;
    Object_68 *obj68;
    s32 sp3C;
    UNUSED s32 pad;
    Object *firstRacerObj;
```

File: src/vehicle_smokey.c [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    }
+}
+
+void func_8005E204(Object *obj, Object_Racer *racer, f32 arg2, s32 objectID, s32 arg4) {
+    Object** var_s6;
+    s32 sp80;
+    s32 sp7C;
+    LevelObjectEntry* temp_s2;
+    Object* temp_s1;
+    Object* newObj;
```
Comment (DavidSM64) [inline src/vehicle_smokey.c:251]:
```suggestion
    Object **var_s6;
    s32 sp80;
    s32 sp7C;
    LevelObjectEntry *temp_s2;
    Object *temp_s1;
    Object *newObj;
```

File: src/vehicle_tricky.c [hunk 5]
```diff
@@ -60,184 +60,180 @@ void func_8005C2F0(Object *object, unk8005C2F0 *arg1) {
...
-    Object_68 *temp_v0_2;
-    Object *temp_v0_4;
+    f32 xDiff;
+    f32 zDiff;
+    ObjectModel *objModel;
+    s32 sp40;
+    Object_68 *obj68;
+    s32 sp38;
+    UNUSED s32 pad;
+    Object* firstRacerObj;
```
Comment (DavidSM64) [inline src/vehicle_tricky.c:77]:
```suggestion
    Object *firstRacerObj;
```

**Review by DavidSM64 — APPROVED** [review]
Besides these asterisks, looks good to me.

File: src/object_functions.c [hunk 6]
```diff
@@ -989,7 +990,7 @@ void obj_loop_unknown58(Object *obj, s32 updateRate) {
     someOtherObj64 = &someOtherObj->unk64->unkid58;
     obj60 = obj->unk60;
     if (obj60->unk0 == 1) {
-        s8 temp = someOtherObj64->unk1D6;
+        s8 temp = someOtherObj64->vehicleID;
         if (temp == 1 || temp == 2) {
```
Comment (Ryan-Myers) [inline src/object_functions.c:994]:
```suggestion
        if (temp == VEHICLE_HOVERCRAFT || temp == VEHICLE_PLANE) {
```

File: src/object_functions.c [hunk 7]
```diff
@@ -3264,15 +3273,15 @@ void obj_loop_bridge_whaleramp(Object *obj, s32 updateRate) {
         racerObj = get_racer_object(0);
         if (racerObj != NULL) {
             racer = (Object_Racer *) racerObj->unk64;
-            switch(racer->unk1D6) {
+            switch(racer->vehicleID) {
                 default:
-                    var_v0 = 1;
+                    var_v0 = VEHICLE_HOVERCRAFT;
                     break;
                 case 1:
```
Comment (Ryan-Myers) [inline src/object_functions.c:3280]:
```suggestion
                case VEHICLE_HOVERCRAFT:
```

File: src/object_functions.c [hunk 8]
```diff
@@ -3264,15 +3273,15 @@ void obj_loop_bridge_whaleramp(Object *obj, s32 updateRate) {
...
-                    var_v0 = 2;
+                    var_v0 = VEHICLE_PLANE;
                     break;
                 case 2:
```
Comment (Ryan-Myers) [inline src/object_functions.c:3283]:
```suggestion
                case VEHICLE_PLANE:
```

File: src/objects.c [hunk 9]
```diff
@@ -1397,7 +1397,7 @@ void render_racer_shield(Gfx **dList, MatrixS **mtx, Vertex **vtxList, Object *o
         if (var_a2 > 10) {
             var_a2 = 0;
         }
-        var_a1 = racer->unk1D6;
+        var_a1 = racer->vehicleID;
         if (var_a1 > 2) {
```
Comment (Ryan-Myers) [inline src/objects.c:1401]:
```suggestion
        if (var_a1 >= NUMBER_OF_PLAYER_VEHICLES) {
```

File: src/objects.c [hunk 10]
```diff
@@ -1468,7 +1468,7 @@ void render_racer_magnet(Gfx **dList, MatrixS **mtx, Vertex **vtxList, Object *o
             gObjectCurrMatrix = *mtx;
             gObjectCurrVertexList = *vtxList;
             magnet = (f32 *) get_misc_asset(MISC_ASSET_MAGNET_DATA);
-            var_a0 = racer->unk1D6;
+            var_a0 = racer->vehicleID;
             if (var_a0 < 0 || var_a0 > 2) {
```
Comment (Ryan-Myers) [inline src/objects.c:1472]:
```suggestion
            if (var_a0 < VEHICLE_CAR || var_a0 >= NUMBER_OF_PLAYER_VEHICLES) {
```

File: src/particles.c [hunk 11]
```diff
@@ -770,9 +769,346 @@ void func_800B03C0(Object *arg0, ObjectTransform *arg1, unk800B03C0_arg2 *arg2,
...
+    f32 temp_f0;
+    f32 temp_f14;
+    f32 temp_f2;
+
+    sp2C = gParticlesAssetTable[arg1->unk8];
+    if (sp2C->unk0 != 4) {
+        return NULL;
+    }
+    sp20 = arg1->behaviour;
+    sp2C = sp2C;
```
Comment (Ryan-Myers) [inline src/particles.c:786]:
```suggestion
```

File: src/vehicle_tricky.c [hunk 12]
```diff
@@ -60,184 +60,180 @@ void func_8005C2F0(Object *object, unk8005C2F0 *arg1) {
...
+            obj->segment.y_velocity += 7.5;
+            func_8005CB04(1);
+            play_sound_global(SOUND_EXPLOSION, NULL);
+            set_camera_shake(12.0f);
+            racer->velocity *= 0.3;
+            racer->unkC = 0.0f;
+        }
     }
-    racer->attackType = ATTACK_NONE;
+    racer->attackType = 0;
```
Comment (Ryan-Myers) [inline src/vehicle_tricky.c:125]:
```suggestion
    racer->attackType = ATTACK_NONE;
```

File: src/vehicle_tricky.c [hunk 13]
```diff
@@ -60,184 +60,180 @@ void func_8005C2F0(Object *object, unk8005C2F0 *arg1) {
...
+    if ((firstRacerObj->segment.trans.y_position + 400.0) < D_8011D5C0) {
+        if (func_800C018C() == 0 && is_in_two_player_adventure()) {
             func_8006F398();
         }
         func_8006F140(1);
     }
-    if ((obj == temp_v0_4->interactObj->obj) && (temp_v0_4->interactObj->unk14 & 8) && (obj->segment.unk38.byte.unk3A == 1)) {
-        temp_s0->attackType = ATTACK_SQUISHED;
+    if (obj == firstRacerObj->interactObj->obj && firstRacerObj->interactObj->unk14 & 8 && obj->segment.unk38.byte.unk3B == 1) {
+        racer->attackType = 4;
```
Comment (Ryan-Myers) [inline src/vehicle_tricky.c:204]:
```suggestion
        racer->attackType = ATTACK_SQUISHED;
```

**Review by Ryan-Myers — APPROVED** [review]
Flying untested with my suggestions. All looks good though.

**Pull request merged**